internal/encoding/encode.go

```diff
@@ -383,6 +383,12 @@ func WriteBinary(wr *buffer.Buffer, bin []byte) error {
 	l := len(bin)
 
 	switch {
+	case l == 0:
```

Comment:
The change seems to make sense compared to what .NET and Java did. I'm struggling to find in the amqp spec that this is the correct behavior.

Fix is good so long as its adhering to the spec.

Reply:
For null definition: https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-null

Previously, we were encoding this as [`vbin8`](https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-binary) with a length of zero. So, we were saying "we have zero bytes of vbin8 data". While I think both are technically correct, encoding it as `null` is slightly more compact and follows what others are doing.

Reply:
Also of interest https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#section-composite-type-representation

" Note that a null value and a zero-length array (with a correct type for its elements) both describe an absence of a value and MUST be treated as semantically identical."

So both are correct, just the `null` is one less byte.

Reply:
Let me check something on this. I recall there was some discussion about this, just want to check compatibility here.

Reply:
I think we should hold this for a moment because it does result in a behavior change for clients for Event Hubs.

In Service Bus this doesn't result in a behavior change for the client. Sending any of these ([]byte{}, nil) results in a `nil` Body (both before and after your change).

In Event Hubs, however, sending any of these ([]byte, nil) results in an `[]byte{}` each time **before** this change, but **after** now returns a `[]byte(nil)`. This might not matter in practice, but I'd still consider it a breaking change.

Reply:
Interesting. This is on the receiving side? Our current behavior is compliant with the spec, so if this is causing a change in behavior then I don't think it's worth changing.

Reply:
Yeah, receiving side.